llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -1690,114 +1702,173 @@ void SpirvLowerGlobal::lowerBufferBlock() {
             if (inst->getFunction() != func)
               continue;
 
-            // We have a user of the global, expect either a GEP or a bitcast.
+            ReplaceInstsInfo replaceInstsInfo = {};
+            // We have a user of the global, expect a GEP, a bitcast or a select.
             if (auto *getElemPtr = dyn_cast<GetElementPtrInst>(inst)) {
-              getElemPtrsToReplace.push_back(getElemPtr);
-            } else {
+              replaceInstsInfo.getElemPtrInsts.push_back(getElemPtr);
+            } else if (auto *bitCast = dyn_cast<BitCastInst>(inst)) {
               // We need to modify the bitcast if we did not find a GEP.
-              BitCastInst *const bitCast = dyn_cast<BitCastInst>(inst);
-              assert(bitCast);
               assert(bitCast->getOperand(0) == &global);
-
-              bitCastsToModify.push_back(bitCast);
+              replaceInstsInfo.bitCastInst = bitCast;
+            } else {
+              // The users of the select must be a GEP.
+              SelectInst *selectInst = dyn_cast<SelectInst>(inst);
+              assert(selectInst);
+              assert(selectInst->getTrueValue() == &global || selectInst->getFalseValue() == &global);
+              replaceInstsInfo.selectInst = selectInst;
+              for (User *selectUser : selectInst->users()) {
+                if (auto *userInst = dyn_cast<Instruction>(selectUser)) {
+                  if (userInst->getFunction() != func)
```

Comment:
I think we needn't check userInst->getFunction() != func for the users of select instruction. please replace it with assert.